### DISCUSSION TITLE
unzip slow

### DISCUSSION THREAD

**Discussion by lifejoyforpy** [icsharpcode/SharpZipLib#780] [Q&A]
Hi，I use unzip from zip file in a linux pod environment, I found slowly (100MiB zip files takes 20s to decompress). My pod use mount azure blob.

The decompression code is the following:
![1664516380142](https://user-images.githubusercontent.com/38183800/193198264-b2e2abe7-d8f0-436b-934d-6354fdc9c468.jpg)

What can I do to improve the unzip performance?

**Comment by piksel** [comment]
I don't see anything obvious, but using managed code (SharpZipLib) to do CPU-heavy tasks will always be slower than native. If speed is an issue you could try using a native executable like `7za`.

**Reply by piksel** [reply]
Although, the buffer you are using is covered by the tooltip. You can try using larger (or smaller) buffer size.
